spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```diff
@@ -0,0 +1,210 @@
+        if (!fieldName.equals(labelField) && (excludeFeatures.isEmpty() || !excludeFeatures.contains(fieldName))) {
+          fields.add(fieldName);
+        }
+      }
+    }
+    return fields;
+  }
+
+  /**
+   * Get the getCategoricalFeatureInfo using the feature to cardinality mapping provided by the user.
+   *
+   * @param inputSchema        schema of the received record.
+   * @param featuresToInclude  features to be used for training/prediction.
+   * @param featuresToExclude  features to be excluded when training/predicting.
+   * @param labelField         field containing the prediction values.
+   * @param cardinalityMapping feature to cardinality mapping specified for categorical features.
+   * @return categoricalFeatureInfo for categorical features.
+   */
+  public static Map<Integer, Integer> getCategoricalFeatureInfo(Schema inputSchema, String featuresToInclude,
+                                                                String featuresToExclude, String labelField,
+                                                                String cardinalityMapping) {
+    List<String> featureList = getFeatureList(inputSchema, featuresToInclude, featuresToExclude, labelField);
+    Map<Integer, Integer> outputFieldMappings = new HashMap<>();
+
+    if (!Strings.isNullOrEmpty(cardinalityMapping)) {
+      for (String field : Splitter.on(',').trimResults().split(cardinalityMapping)) {
+        String[] value = field.split(":");
+        if (value.length == 2) {
+          try {
+            outputFieldMappings.put(featureList.indexOf(value[0]), Integer.parseInt(value[1]));
+          } catch (NumberFormatException e) {
+            throw new IllegalArgumentException(
+              String.format("Invalid cardinality %s. Please specify valid integer for cardinality", value[1]));
+          }
+        } else {
+          throw new IllegalArgumentException(
+            String.format("Invalid categorical feature mapping '%s'. Please make sure it is in the format " +
+                            "'feature':'cardinality'.", field));
+        }
+      }
+    }
+    return outputFieldMappings;
+  }
+
+  /**
+   * Validate prediction field value for Predictor.
+   *
+   * @param inputSchema     schema of the received record.
+   * @param predictionField field on which to set the prediction.
+   */
+  public static void validatePredictionFieldForRegressor(Schema inputSchema, String predictionField) {
+    Preconditions.checkArgument(inputSchema.getField(predictionField) == null, "Prediction field must not already " +
+      "exist in the input schema.");
+  }
+
+  /**
+   * Validate label field for trainer.
+   *
+   * @param inputSchema schema of the received record.
+   * @param labelField  field from which to get the prediction.
+   */
+  public static void validateLabelFieldForTrainer(Schema inputSchema, String labelField) {
+    Schema.Field prediction = inputSchema.getField(labelField);
+    if (labelField == null) {
```

Comment:
For trainer, we are using the term labelField instead of predictionField as is mentioned in the design discussion https://wiki.cask.co/display/CE/Decision+Tree+Regression.
Please let me know your thoughts.